homeassistant/components/vicare/__init__.py

```diff
@@ -115,6 +116,31 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     return unload_ok
 
 
+async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    if entry.version == 1:
+        if entry.minor_version == 1:
+            _LOGGER.debug(
+                "Migrating from version %s.%s", entry.version, entry.minor_version
+            )
+            hass.config_entries.async_update_entry(
+                entry,
+                minor_version=2,
+                data={
+                    CONF_CLIENT_ID: entry.data.get(CONF_CLIENT_ID),
+                    CONF_USERNAME: entry.data.get(CONF_USERNAME),
+                    CONF_PASSWORD: entry.data.get(CONF_PASSWORD),
+                },
+                options={CONF_HEATING_TYPE: entry.data.get(CONF_HEATING_TYPE)},
+            )
+            _LOGGER.debug(
+                "Migration to version %s.%s successful",
+                entry.version,
+                entry.minor_version,
+            )
+    return True
```

Comment:
Should have here to check we are not downgrading from a future major version
Ex: https://github.com/home-assistant/core/blob/dev/homeassistant/components/dnsip/__init__.py#L34-L36